src/nvim/option.c

```diff
@@ -1142,8 +1142,18 @@ static OptVal get_option_newval(int opt_idx, int opt_flags, set_prefix_T prefix,
 {
   assert(varp != NULL);
 
+  vimoption_T *opt = &options[opt_idx];
   char *arg = *argp;
-  OptVal oldval = optval_from_varp(opt_idx, varp);
+  void *oldval_varp = NULL;
+
+  // When setting the local value of a global option, the old value may be the global value.
+  if (((int)opt->indir & PV_BOTH) && (opt_flags & OPT_LOCAL)) {
+    oldval_varp = get_varp(opt);
+  } else {
+    oldval_varp = varp;
+  }
```

Comment:
this looks like a ternary.

Reply:
I did try making this a ternary but imho it looks confusing as a ternary. Plus it wasn't a ternary in the original code iirc.